teamcity.build

```diff
@@ -245,231 +246,278 @@
 			<import namespace="System.Data.Common"/>
 			<import namespace="System.Data.SqlClient"/>
 			<import namespace="System.IO"/>
+			<import namespace="System.Linq"/>
 			<import namespace="System.Xml"/>
 		</imports>
 		<code>
 			<![CDATA[
 
-						public static void StripAttributes(XmlDocument source, string xpath)
+		public static void StripAttributes(XmlDocument source, string xpath)
+		{
+			foreach (XmlAttribute att in source.SelectNodes(xpath))
+				att.OwnerElement.RemoveAttribute(att.Name);
+		}
+
+		public static void StripElements(XmlDocument source, string xpath)
+		{
+			foreach (XmlElement el in source.SelectNodes(xpath))
+				el.ParentNode.RemoveChild(el);
+		}
+
+		[Function("StripTimings")]
+		public static string StripTimings(string testResultFile, string outputFile)
+		{
+			try
+			{
+				XmlDocument testResults = new XmlDocument();
+				testResults.Load(testResultFile);
+
+				StripElements(testResults, "//stack-trace");
+				StripAttributes(testResults, "//@cwd");
+				StripAttributes(testResults, "//@total");
+				StripAttributes(testResults, "//@date");
+				StripAttributes(testResults, "//@time");
+				StripAttributes(testResults, "//@machine-name");
+				StripAttributes(testResults, "//@user");
+				StripAttributes(testResults, "//@user-domain");
+
+				outputFile = Path.GetDirectoryName(testResultFile) + "/" + outputFile;
+				testResults.Save(outputFile);
+				return outputFile;
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine(e);
+				throw;
+			}
+		}
+
+		public enum ResultStatus
+		{
+			None,
+			Ignored,
+			Success,
+			Failure
+		}
+
+		public class Result
+		{
+			public string Name;
+			public ResultStatus Status;
+			public bool Executed;
+			public bool Success;
+
+			public bool Inconclusive
+			{
+				get
+				{
+					return Executed && !Success && Status == ResultStatus.Success;
```

Comment:
Issuing a NUnit warning causes the test to have `Success` as `result` attribute value with `false`  as `success` attribute value. So I have added an `Inconclusive` property matching these on the `Result` class.